File: sein/22_12_28_01.java
/*
행렬의 덧셈은 행과 열의 크기가 같은 두 행렬의 같은 행, 같은 열의 값을 서로 더한 결과가 됩니다. 
2개의 행렬 arr1과 arr2를 입력받아, 행렬 덧셈의 결과를 반환하는 함수, solution을 완성해주세요.
*/

class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr1[0].length];
        int a = 0;
        int b = 0;
        
        for(int i=0; i<arr2.length; i++){               
            for(int j=0; j<arr2[0].length; j++){           
                a = arr1[i][j];                             
                b = arr2[i][j];                     
                answer[i][j] = a+b;                 
            }
        }
        return answer;
    }
}

File: yejin/22_12_28_01.py
# 행렬의 덧셈

# 시도1
def solution(arr1, arr2):
    answer =[]
    for i in range(len(arr1)):
        for j in range(len(arr1[0])):
            arr1[i][j]+=arr2[i][j]
            answer.append(arr1[i][j])
    return answer
# 문제: 실행한 결괏값 [4,6,7,9]이 기댓값 [[4,6],[7,9]]과 다릅니다.
# 해결: 이중배열로 만들기

# 시도2
def solution(arr1, arr2):
    answer =[]
    for i in range(len(arr1)):
        a = []
        for j in range(len(arr1[0])):
            a.append(arr1[i][j]+arr2[i][j])
        answer.append(a)
    return answer

# arr1의 길이만큼 반복문을 돌려줌.
# 이중 반복문을 만들어 열의 길이만큼 돌려주며 값을 더해줌.
# 더한 값을 a에 붙여넣어주고 열이 다 돌아가면 answer에 붙여줌.
# 이중배열을 만들어 return해줌.
